src/GitReleaseManager.Core/Provider/GitLabProvider.cs

```diff
@@ -0,0 +1,435 @@
+        {
+            return ExecuteAsync(async () =>
+            {
+                var projectId = GetGitLabProjectId(owner, repository);
+
+                if (issue.IsPullRequest)
+                {
+                    var mergeRequestClient = _gitLabClient.GetMergeRequest(projectId);
+                    var commentsClient = mergeRequestClient.Comments(issue.PublicNumber);
+                    var mergeRequestComment = new MergeRequestCommentCreate
+                    {
+                        Body = comment,
+                    };
+
+                    commentsClient.Add(mergeRequestComment);
+                }
+                else
+                {
+                    var issueNotesClient = _gitLabClient.GetProjectIssueNoteClient(projectId);
+                    var issueComment = new ProjectIssueNoteCreate
+                    {
+                        IssueId = issue.PublicNumber,
+                        Body = comment,
+                    };
+
+                    issueNotesClient.Create(issueComment);
+                }
+            });
+        }
+
+        public Task<IEnumerable<Issue>> GetIssuesAsync(string owner, string repository, Milestone milestone, ItemStateFilter itemStateFilter = ItemStateFilter.All)
+        {
+            return ExecuteAsync(async () =>
+            {
+                var issuesClient = _gitLabClient.Issues;
+
+                var query = new IssueQuery();
+                query.Milestone = milestone.Title;
+
+                if (itemStateFilter == ItemStateFilter.Open)
+                {
+                    query.State = IssueState.opened;
+                }
+                else if (itemStateFilter == ItemStateFilter.Closed)
+                {
+                    query.State = IssueState.closed;
+                }
+
+                var projectId = GetGitLabProjectId(owner, repository);
+
+                var issues = issuesClient.GetAsync(projectId, query);
+
+                var mergeRequestsClient = _gitLabClient.GetMergeRequest(projectId);
+
+                var mergeRequestQuery = new MergeRequestQuery();
+                mergeRequestQuery.Milestone = milestone.Title;
+
+                if (itemStateFilter == ItemStateFilter.Open)
+                {
+                    mergeRequestQuery.State = MergeRequestState.opened;
+                }
+                else if (itemStateFilter == ItemStateFilter.Closed)
+                {
+                    mergeRequestQuery.State = MergeRequestState.merged;
+                }
+
+                var mergeRequests = mergeRequestsClient.Get(mergeRequestQuery);
+
+                var issuesAndMergeRequests = new List<Issue>();
+                issuesAndMergeRequests.AddRange(_mapper.Map<IEnumerable<Issue>>(issues));
+                issuesAndMergeRequests.AddRange(_mapper.Map<IEnumerable<Issue>>(mergeRequests));
+
+                return issuesAndMergeRequests.AsEnumerable();
```

Comment:
```suggestion
                return Task.FromResult(issuesAndMergeRequests.AsEnumerable());
```

May not work. It may be that you need to use the explicit return result (not entirely sure).